### DISCUSSION TITLE
Visually seeing what the bots thoughts are

### DISCUSSION THREAD

**Discussion by brogers-tech** [SebLague/Chess-Challenge#333] [Ideas]
Was wondering if it was possible to gain access to the boardUI so the bot, as it is thinking, can color the squares it is considering or do some other visualizations.  This would be great for debugging and even verifying optimizations are being made.  It would be cool to do something similar to this old webpage

[https://www.bewitched.com/chess/](https://www.bewitched.com/chess/)

(Don't use the webpage for testing as it has some flaws in its gameplay logic such as allowing a king to castle even when it is in check).

**Comment by 101dal** [comment]
It's a good idea at the start but it should be only an option (maybe you can like encrypt data and print it in the logs and then use an external program to visualize to avoid slowing down the computation since it's already slow enough :) )

**Reply by brogers-tech** [reply]
Yeah, I agree it would slow down the initial checks but it might help with debugging to see what your bot is doing.  Of course, you would have to add the code to your bot to change the square colors while it is evaluating the position.  That's more of what I was hoping for.  Also, it would just be cool to see.  :D
